components/raftstore/src/coprocessor/split_check/mod.rs

```diff
@@ -85,8 +85,40 @@ impl<'a, E> Host<'a, E> {
         Ok(vec![])
     }
 
+    pub fn approximate_bucket_keys<Kv: engine_traits::KvEngine>(
+        &mut self,
+        region: &Region,
+        engine: &Kv,
+    ) -> Result<Vec<Vec<u8>>> {
+        let region_size = get_region_approximate_size(engine, region, 0)?;
+        const MIN_BUCKET_COUNT_PER_REGION: u64 = 2;
+        if region_size >= self.cfg.region_bucket_size.0 * MIN_BUCKET_COUNT_PER_REGION {
+            let mut bucket_checker = size::Checker::new(
+                self.cfg.region_bucket_size.0, /* not used */
+                self.cfg.region_bucket_size.0, /* not used */
+                region_size / self.cfg.region_bucket_size.0,
+                CheckPolicy::Approximate,
+            );
+            let keys = box_try!(bucket_checker.approximate_split_keys(region, engine));
+            if !keys.is_empty() {
+                return Ok(keys);
+            }
```

Comment:
```suggestion
            return bucket_checker.approximate_split_keys(region, engine);
```

Reply:
good catch. I followed the pattern somewhere else, but apparently that was in a loop.